src/main/java/com/epam/Main.java

```diff
@@ -1,24 +1,171 @@
+ 
+    public static void main(String[] args) throws IOException, ExecutionException, InterruptedException {
+        String[] inputFilesList = args[0].split(",");
+        LinkedList<String> inputFileQueue = new LinkedList<String>();
+        inputFileQueue.addAll(Arrays.asList(inputFilesList));
+        
+        int threadPoolSize = 0;
+        try{
+        	threadPoolSize = Integer.parseInt(args[1]);
+        }catch(Exception e){
+        	threadPoolSize = DEFAULT_THREAD_POOL_SIZE;
+        }
+        
+        int batchSize = 0;
+        try{
+        	batchSize = Integer.parseInt(args[2]);
+        }catch(Exception e){
+        	batchSize = DEFAULT_BATCH_SIZE;
+        }
+        
+        int producerSize = 0;
+        try{
+        	producerSize = Integer.parseInt(args[3]);
+        }catch(Exception e){
+        	producerSize = DEFAULT_PRODUCER_SIZE; 
+        }
+        
+        int consumerSize = 0;
+        try{
+        	consumerSize = Integer.parseInt(args[4]);
+        }catch(Exception e){
+        	consumerSize = DEFAULT_CONSUMER_SIZE; 
+        }
+        handleMultipleFilesWithCallable(inputFileQueue,threadPoolSize,batchSize,producerSize,consumerSize);
+    }
+    
+    public static void handleMultipleFilesWithCallable(LinkedList<String> inputFileQueue,int threadPoolSize, int batchSize,int producerSize, int consumerSize) throws ExecutionException, InterruptedException, IOException {
+        
+        ExecutorService executor = Executors.newFixedThreadPool(threadPoolSize);
 
-    private static final String ACCIDENTS_CSV = "src/main/resources/DfTRoadSafety_Accidents_2009.csv";
-
-
-    public static void main(String[] args) throws IOException {
-        AccidentsDataLoader accidentsDataLoader = new AccidentsDataLoader();
-        List<RoadAccident> accidents = accidentsDataLoader.loadRoadAccidents(ACCIDENTS_CSV);
+        BlockingQueue<List<RoadAccident>> dataQueue = new ArrayBlockingQueue<List<RoadAccident>>(DEFAULT_STORAGE_QUEUE_SIZE);        
+        FileWriter fileWriter1 = new FileWriter(OUTPUT_FILE_PATH_DAYTIME);
+        CSVPrinter csvWriterDayTime = new CSVPrinter(fileWriter1,csvFormat);
+        printCSVHeader(csvWriterDayTime);
+        
+        FileWriter fileWriter2 = new FileWriter(OUTPUT_FILE_PATH_NIGHTTIME);
+        CSVPrinter csvWriterNightTime = new CSVPrinter(fileWriter2,csvFormat);
+        printCSVHeader(csvWriterNightTime);
+        //start consumer threads
+        System.out.println("Starting Consumers");
+        for(int i=0;i<consumerSize;i++){
+        	executor.execute(new DataCSVFileWriter(dataQueue, csvWriterDayTime,csvWriterNightTime));
+        }    
+            
+       
 
+        //Create 2 reader task and start
+        System.out.println("Starting readers");
+        List<FutureTask<Integer>> futureTaskList = new ArrayList<FutureTask<Integer>>(); 
+        
+        for (int i = 0; i < producerSize; i++) {
```

Comment:
looks like producerSize should be exectly equal to size of inputFileQueue. 
You use remove() method - so you run one task per each element in inputFileQueue list. If producerSize less than the list - you will just ignore some files, right? 

BTW - why do you call inputFileQueue the variable of List type? :confused: